pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -107,13 +107,20 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 			Dependencies: flow.NewTaskIDs(deployNamespace),
 		})
 		deployKubeAPIServerService = g.Add(flow.Task{
-			Name:         "Deploying Kubernetes API server service in the Seed cluster",
-			Fn:           flow.TaskFn(botanist.DeployKubeAPIServerService).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
+			Name: "Deploying Kubernetes API server service in the Seed cluster",
+			Fn: flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerService.Deploy).
+				RetryUntilTimeout(defaultInterval, defaultTimeout).
+				SkipIf(o.Shoot.HibernationEnabled && !useSNI),
 			Dependencies: flow.NewTaskIDs(deployNamespace),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying Kubernetes API server service SNI settings in the Seed cluster",
+			Fn:           flow.TaskFn(botanist.DeployKubeAPIServerSNI).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(deployKubeAPIServerService),
```

Comment:
This dependency seems fake to me.
If SNI is enabled, `deployKubeAPIServerService` will be skipped and nothing will be changed.

Reply:
The check is `o.Shoot.HibernationEnabled && !useSNI` - it's skipped when the cluster should be hibernated and it doesn't use SNI

Reply:
I do not see any `SkipIf/DoIf` in this task.

Reply:
I was referring to the API service check which is 

```go
deployKubeAPIServerService = g.Add(flow.Task{
	Name: "Deploying Kubernetes API server service in the Seed cluster",
	Fn: flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerService.Deploy).
		RetryUntilTimeout(defaultInterval, defaultTimeout).
		SkipIf(o.Shoot.HibernationEnabled && !useSNI),
	Dependencies: flow.NewTaskIDs(deployNamespace),
})
```

Reply:
Hm, @vpnachev seems to be right. Why is this task depending on the `deployKubeAPIServerService`? Shouldn't it rather depend on `deployNamespace`?

Reply:
> Hm, @vpnachev seems to be right. Why is this task depending on the `deployKubeAPIServerService`? Shouldn't it rather depend on `deployNamespace`?

The `ClusterIP` of the `kube-apiserver` `Service` is need for creation of the SNI. See https://github.com/gardener/gardener/blob/3f0c10df007972e96117b2e2bd6c30e8d0e8bef8/pkg/operation/botanist/controlplane.go#L1200-L1223 and https://github.com/gardener/gardener/blob/3f0c10df007972e96117b2e2bd6c30e8d0e8bef8/pkg/operation/botanist/controlplane.go#L1244-L1269

Reply:
Thanks for the clarification!